com.regnosys.rosetta.lib/src/main/java/com/rosetta/model/lib/expression/ExpressionOperators.java

```diff
@@ -262,4 +268,30 @@ private static <T> String formatMultiError(Mapper<T> o) {
 				t.getClass().getSimpleName() : // for rosettaModelObjects only log class name otherwise error messages are way too long
 				o.getMulti().toString();
 	}
+	
+	// one-of and choice
+
+	public static <T> ComparisonResult choice(Mapper<T> mapper, List<String> choiceFieldNames, ValidationResult.ChoiceRuleValidationMethod necessity) {
+		T object = mapper.get();
+		List<String> populatedFieldNames = new LinkedList<>();
+		for (String a: choiceFieldNames) {
+			try {
+				Method getter = object.getClass().getMethod("get" + StringUtils.capitalize(a));
+				if (ExistenceChecker.isSet(getter.invoke(object))) {
+					populatedFieldNames.add(a);
+				}
+			} catch (NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+				throw new IllegalArgumentException(e);
```

Comment:
What is the impact of throwing here? Does the code gen need to handle `IllegalArgumentException`?

Reply:
This won't throw during code gen, but it can throw when using generated code.

However, it'll only throw if you actually provide a nonsense argument. For example, if you have a type with attributes called `foo` and `bar`, then you should only pass in these attribute names. If you pass in `party`, this will throw.

This will of course never happen in the code we generate. It's only when a consumer of our generated code tries to do something wrong that this will throw.